backend/simplemap/simplemap.go

```diff
@@ -0,0 +1,56 @@
+package simplemap
+
+import (
+	"time"
+	"fmt"
+	"strconv"
+	"github.com/heetch/confita/backend"
+	"context"
+)
+
+// Backend that loads config from map
+type Backend struct{
+	theMap	map[string]interface{}
+}
+
+// NewBackend creates a simplemap backend.
+func NewBackend(theMap map[string]interface{}) *Backend {
+	return &Backend{
+		theMap: theMap,
+	}
+}
+
+// Get loads the given key from the map
+func (b *Backend) Get(ctx context.Context, key string) ([]byte, error) {
+	val := b.theMap[key]
+	if val == nil {
+		return nil, backend.ErrNotFound
+	}
+
+	if str, ok := val.(string); ok {
+		return []byte(str), nil
+	}
+
+	if u, ok := val.(uint); ok {
+		return []byte(strconv.FormatUint(uint64(u), 10)), nil
+	}
+
+	if i, ok := val.(int); ok {
+		return []byte(strconv.FormatInt(int64(i), 10)), nil
+	}
+
+	if f, ok := val.(float64); ok {
+		return []byte(fmt.Sprintf("%g", f)), nil
+	}
+
+	if d, ok := val.(time.Duration); ok {
+		return []byte(fmt.Sprintf("%s", d)), nil
+	}
+
+	return nil, backend.ErrNotFound
+}
```

Comment:
I think that this is redundant with what we do in the [convert](https://github.com/heetch/confita/blob/master/config.go#L230-L299) function. Moreover you are not handling all the primitive and this is very problematic, for example if the type is an `int64` the method will end with a `backend.ErrNotFound` error.
What can you do to improve this ? 
There is also the [Unmarshaler](https://github.com/heetch/confita/blob/master/config.go#L308-L310) and [StructLoader](https://github.com/heetch/confita/blob/master/config.go#L313-L315) interface which can be used maybe you can take a look at them (it's just a proposal not sure that is the correct way to go).

Reply:
Thank you! Tried the StructLoader way, take a look :)